integrations/influxdb_connector/influxdb-timestream-connector/src/lib.rs

```diff
@@ -17,13 +24,20 @@ pub mod timestream_utils;
 // that can be made per second is 1.
 pub static TIMESTREAM_API_WAIT_SECONDS: u64 = 1;
 
+// The number of batches processed at the same time.
+// For multi-table multi measure schema, batches are a combination of
+// a table name and a Vec of records bound for that table
+pub static NUM_BATCH_THREADS: usize = 16;
+
 async fn handle_body(
-    client: &timestream_write::Client,
+    client: &Arc<timestream_write::Client>,
     body: &[u8],
     precision: &timestream_write::types::TimeUnit,
 ) -> Result<(), Error> {
     // Handle parsing body in request
 
+    let function_start = Instant::now();
```

Comment:
Using tracing subscriber would be a more flexible and clean solution for logging execution time. Rather than having to initialize a timer, we can use the `tracing::instrument` macro. Here is a quick example:

```toml
[dependencies]
tracing = "0.2"
tracing-subscriber = "0.3"

```

```rust
use std::time::Duration;
use tracing_subscriber::fmt;
use tracing_subscriber::fmt::format::FmtSpan;

#[tracing::instrument]
async fn hoo() {
    std::thread::sleep(Duration::from_millis(100));
}

#[tracing::instrument]
async fn blah() {
    std::thread::sleep(Duration::from_millis(200));
    hoo().await;
}

#[tokio::main]
async fn main() {
    fmt::fmt()
        .with_span_events(FmtSpan::CLOSE)
        .with_target(false)
        //.with_max_level(tracing::Level::TRACE)
        .with_level(false)
        .init();

    for _ in 0..3 {
        blah().await;
    }
}
```

Reply:
Good idea, I have changed this to use `tracing-subscriber`. The tracing subscriber will be used for all logs, the logging of function times are only enabled when `RUST_LOG` is `TRACE`. Here is a sample output, with `compact` formatting

```shell
. . .
2024-10-22T19:36:35.177829Z TRACE                 main Lambda runtime invoke:lambda_handler:handle_body:handle_multi_table_ingestion: influxdb_timestream_connector: close time.busy=3.86ms time.idle=1.14s requestId="db9350f1-acdb-48f1-a9fa-a2a128f962ab" xrayTraceId="Root=1-6717fec1-b512bb1a95a756fb41e8d3b3;Parent=b6a137768f7989d8;Sampled=1"
2024-10-22T19:36:35.178068Z TRACE                 main Lambda runtime invoke:lambda_handler:handle_body: influxdb_timestream_connector: close time.busy=280ms time.idle=1.14s requestId="db9350f1-acdb-48f1-a9fa-a2a128f962ab" xrayTraceId="Root=1-6717fec1-b512bb1a95a756fb41e8d3b3;Parent=b6a137768f7989d8;Sampled=1"
2024-10-22T19:36:35.178079Z TRACE                 main Lambda runtime invoke:lambda_handler: influxdb_timestream_connector: close time.busy=280ms time.idle=1.14s requestId="db9350f1-acdb-48f1-a9fa-a2a128f962ab" xrayTraceId="Root=1-6717fec1-b512bb1a95a756fb41e8d3b3;Parent=b6a137768f7989d8;Sampled=1"
```